src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmBrowserTestRunner.cs

```diff
@@ -164,7 +164,14 @@ private async Task RunConsoleMessagesPump(WebSocket socket, TaskCompletionSource
                         var line = Encoding.UTF8.GetString(mem.GetBuffer(), 0, (int)mem.Length);
                         line += Environment.NewLine;
 
-                        _messagesProcessor.Invoke(line);
+                        try
+                        {
+                            _messagesProcessor.Invoke(line);
+                        }
+                        catch (Exception ex) when (_messagesProcessor.WasmExitReceivedTcs.Task.IsCompletedSuccessfully)
```

Comment:
Don't we need the same fix for `RunSeleniumLogMessagePump()` ?

I feel like this is just papering over the real issue, which is that the WasmTestMessagesProcessor is disposed before we're finished writing.

Reply:
> Don't we need the same fix for RunSeleniumLogMessagePump() ?

hm yeah

> I feel like this is just papering over the real issue, which is that the WasmTestMessagesProcessor is disposed before we're finished writing.

It is! I'm trying to get us to passing test runs on CI, ASAP, while trying to collect info to debug what might be causing this. The traces from the current runs were not helpful.

Reply:
I think that we were getting that error because we had selenium logging messages, and we would get some after the test was done, as selenium was closing the session. Since, now we are logging to a file instead, we don't get those messages. But trying to be careful anyway, and handle the error if, and when it pops up!

I think this might help too, but that's just based on reading the code, and not traces etc!
```diff
diff --git a/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmTestMessagesProcessor.cs b/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmTestMessagesProcessor.cs
index ccfd00a..7803f6a 100644
--- a/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmTestMessagesProcessor.cs
+++ b/src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/WasmTestMessagesProcessor.cs
@@ -99,9 +99,11 @@ namespace Microsoft.DotNet.XHarness.CLI.Commands.Wasm
             {
                 if (line.Contains("ENDRESULTXML"))
                 {
-                    _xmlResultsFileWriter.Flush();
-                    _xmlResultsFileWriter.Dispose();
+                    var oldWriter = _xmlResultsFileWriter;
                     _xmlResultsFileWriter = null;
+
+                    oldWriter.Flush();
+                    oldWriter.Dispose();
                     return;
                 }
```